ui/component/or-rules/src/or-rule-tree.ts

```diff
@@ -127,7 +127,11 @@ export class OrRuleTree extends OrTreeMenu {
             if (!this.rules) {
                 this._loadRulesets(this.global).then(rulesets => this.rules = rulesets);
             } else {
-                this.nodes = [...this._getRuleNodes(this.rules), ...this._cachedEmptyGroupNodes];
+                const ruleNodes = this._getRuleNodes(this.rules);
+                const ruleNodeIds = new Set(ruleNodes.map(node => node.id));
+                const emptyGroups = this._cachedEmptyGroupNodes
+                    .filter(cachedNode => !ruleNodeIds.has(cachedNode.id));
+                this.nodes = [...ruleNodes, ...emptyGroups];
```

Comment:
I was curious why this code was necessary, so I pulled your branch to my local system.
To my understanding, it's similar to line 138, where groups without any rules are filtered out.

Can you add a short comment in code, so it's clear why/what it's meant to do?